src/NServiceBus.NHibernate/Outbox/OutboxCleaner.cs

```diff
@@ -60,53 +60,43 @@ protected override Task OnStart(IMessageSession busSession)
                 Logger.InfoFormat("Outbox cleanup task is disabled.");
             }
 
-            cleanupTimer = new Timer(PerformCleanup, null, frequencyToRunDeduplicationDataCleanup, Timeout.InfiniteTimeSpan);
+            cancellationTokenSource = new CancellationTokenSource();
 
-            return Task.FromResult(true);
+            cleanup = Task.Run(() => PerformCleanup(cancellationTokenSource.Token), CancellationToken.None);
+
+            return Task.CompletedTask;
         }
 
         protected override Task OnStop(IMessageSession busSession)
         {
-            using (var waitHandle = new ManualResetEvent(false))
-            {
-                cleanupTimer.Dispose(waitHandle);
-
-                waitHandle.WaitOne();
-            }
-
-            return Task.FromResult(true);
+            cancellationTokenSource.Cancel();
+            return cleanup;
         }
 
-        void PerformCleanup(object state)
+        async Task PerformCleanup(CancellationToken ct)
         {
-            try
-            {
-                outboxPersister.RemoveEntriesOlderThan(DateTime.UtcNow - timeToKeepDeduplicationData);
-                circuitBreaker.Success();
-            }
-            catch (Exception ex)
-            {
-                circuitBreaker.Failure(ex);
-            }
-            finally
+            while (ct.IsCancellationRequested == false)
             {
                 try
                 {
-                    cleanupTimer.Change(frequencyToRunDeduplicationDataCleanup, Timeout.InfiniteTimeSpan);
+                    await outboxPersister.RemoveEntriesOlderThan(DateTime.UtcNow - timeToKeepDeduplicationData).ConfigureAwait(false);
```

Comment:
What was the logic of the cleaner before? Did it stop after the first usage? This changes it to an asynchronous poller with a fixed delay for every spin `frequencyToRunDeduplicationDataCleanup`